### ISSUE TITLE
twisted.test.test_sslverify.OpenSSLOptionsTests.test_inspectCertificate doesn't work on Debian 8

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7908]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7908|
|Type|defect|
|Created|2015-05-21 09:36:39Z|
|Branch|https://github.com/twisted/twisted/tree/inspect-certificate-7908|

```
Traceback (most recent call last):
  File "/buildslave/debian8-py2.7/Twisted/twisted/test/test_sslverify.py", line 890, in test_inspectCertificate
    "Public Key with Hash: 50d4b8070143375b3330dedf3a8b9e91"])
  File "/buildslave/debian8-py2.7/Twisted/twisted/trial/_synctest.py", line 437, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/usr/lib/python2.7/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python2.7/unittest/case.py", line 742, in assertListEqual
    self.assertSequenceEqual(list1, list2, msg, seq_type=list)
  File "/usr/lib/python2.7/unittest/case.py", line 724, in assertSequenceEqual
    self.fail(msg)
twisted.trial.unittest.FailTest: Lists differ: ['Certificate For Subject:', '... != ['Certificate For Subject:', '...

First differing element 20:
Public Key with Hash: 50095e748f9bee6128cc94bf2e5c2051
Public Key with Hash: 50d4b8070143375b3330dedf3a8b9e91

Diff is 954 characters long. Set self.maxDiff to None to see it.

twisted.test.test_sslverify.OpenSSLOptionsTests.test_inspectCertificate
```

This is the only thing preventing debian8-py27 being a supported builder.

<details><summary>Searchable metadata</summary>

```
trac-id__7908 7908
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_inspect_certificate_7908 branches/inspect-certificate-7908
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1432200999014316 1432200999014316
changetime__1432204612911508 1432204612911508
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [44899]) Branching to inspect-certificate-7908.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph removed owner|
|-|-|

I think this should work.  Currently supported builders look OK.

Bonus build: https://buildbot.twistedmatrix.com/builders/debian8-py2.7/builds/5

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl set owner to @glyph|
|-|-|

Hi Glyph, looks good, seems to work as designed.

Please add a news fragment and merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

(oh also please remove the now useless comment)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set status to `closed`|
|-|-|

(In [44903]) Merge inspect-certificate-7908: don't hard-code unpredictable keyHash() in inspectCertificate test

Author: glyph

Reviewer: hawkowl

Fixes: [#7908](https://github.com/twisted/twisted/issues/7908)

This is necessary to support Debian 8 as a platform, since Debian 8 has
an OpenSSL which gives (validly) distinct results for this method.  We
should still upgrade public key hashing at some point in the future,
perhaps when cryptography offers an API for it, so we don't need this
workaround forever.
